index.js

```diff
@@ -1,10 +1,12 @@
 import * as cborg from 'cborg'
 import CID from 'multiformats/cid'
-import { codec } from 'multiformats/codecs/codec'
+import { Decoder, Encoder } from 'multiformats/codecs/codec'
 
 // https://github.com/ipfs/go-ipfs/issues/3570#issuecomment-273931692
 const CID_CBOR_TAG = 42
+/** @type {number} */
```

Comment:
We want to use actual literal type here, so TS won't generalize it to number.

```suggestion
/** @type {0x71} */
```

Note: This is not needed if `codec({ code: 0x71, ...` is used as TS can infer that.

Reply:
now that's a weird annotation!

Reply:
Without the annotation it also isn't generalaized. @Gozala I misunderstood you, I thought you want that generalization.

Reply:
I’m pretty sure TS generalizes type of code in `export const code = 0x71` to number.

it did not generalized it in original code `codec({ code: 0x71, ... })` because `codec` user generics.

I apologize for confusion, but I do want non generalized types as it helps with compositions allowing codecs to reject incompatible blocks at compile time

Reply:
I see. The code (without the generalization) produces:

```ts
export const name: "dag-cbor";
export const code: 113;
```

which I thought is a sign that it would also use those specific types directly and not the generalized ones. The reason why I though it is that the generalized version produces:

```ts
/** @type {string} */
export const name: string;
/** @type {number} */
export const code: number;
```

which is clearly generalized.

@Gozala I just checked it with your proposed type annotations, it produces:

```ts
/** @type {'dag-cbor'} */
export const name: 'dag-cbor';
/** @type {0x71} */
export const code: 0x71;
```

Isn't that the same as the non-generalized version?

Anyway, please tell me which version, the one without any type annotion or the one with your type annotation, would be preferred and I'll push an updated version.